### PR TITLE
Add UCesiumFlyToComponent

### LINKED ISSUE #1155: Add UCesiumFlyToComponent
`AGlobeAwareDefaultPawn` has useful methods for doing animated flights to other locations on the globe. We should break this functionality out into a separate `UCesiumFlyToComponent` so that it can be used with any Actor, not just the Cesium-provided pawn. This would allow flying other types of pawns, as well as arbitrary other objects.

### PR DESCRIPTION
**Pull request by kring**
Fixes #1155

This takes the "flight" functionality that was previously in GlobeAwareDefaultPawn and makes it into a component that can be added to anything. So you can do flights with other pawns, or even fly random models around the world or whatever.

Backward compatibility was tricky, but basically Blueprints using the old flight functionality on GlobeAwareDefaultPawn should continue to work (with warnings). C++ code will need changes. DynamicPawn includes a CesiumFlyToComponent. A GlobeAwareDefaultPawn saved in previous versions will gain one when it's upgraded, but users will have to add it manually to new GlobeAwareDefaultPawns.

Other changes here:

* Added `CesiumActorComponentWithGlobeAnchor`, which is just a convenience base class for ActorComponents that require that a GlobeAnchor also be attached to the same Actor. CesiumOriginShiftComponent and CesiumFlyToComponent both derive from this.
* Renamed the various curve assets used in the flights to have more descriptive names. Redirectors should upgrade references to the old names.
* Removed the very old `FloatingPawn` that has been deprecated for over two years now.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @kring!

* :heavy_check_mark: Signed CLA found.

Reviewers, don't forget to make sure that:

- [ ] [Cesium for Unreal Samples](https://github.com/CesiumGS/cesium-unreal-samples) works.

**Review by j9liu — COMMENTED** [review]
Looks great @kring ! All of my comments are doc / style related, so I can push commits for them and save you time. If you disagree with any of my changes feel free to `git revert`.

File: Source/CesiumRuntime/Public/CesiumActorComponentWithGlobeAnchor.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+#pragma once
+
+#include "Components/ActorComponent.h"
+#include "CesiumActorComponentWithGlobeAnchor.generated.h"
+
+class UCesiumGlobeAnchorComponent;
+
+UCLASS(ClassGroup = "Cesium", Abstract)
+class CESIUMRUNTIME_API UCesiumActorComponentWithGlobeAnchor
```
[the remark this answers is not in the document]
Comment (j9liu) [reply Source/CesiumRuntime/Public/CesiumActorComponentWithGlobeAnchor.h]: `UCesiumActorWithGlobeAnchorComponent` was another alternative, but I went with `UCesiumGlobeAnchoredActorComponent`. But feel free to disagree if you think the new names are inaccurate.

**Comment by kring** [comment]
All your changes look great to me. Thanks for the review and fixes @j9liu!

**Comment by kring** [comment]
I'm merging this for the v2.0 preview. Further review comments still welcome.

**Pull request merged**